homeassistant/components/lock/verisure.py

```diff
@@ -0,0 +1,96 @@
+"""
+homeassistant.components.lock.verisure
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Interfaces with Verisure locks.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/verisure/
+"""
+import logging
+
+import homeassistant.components.verisure as verisure
+from homeassistant.components.lock import LockDevice
+
+from homeassistant.const import (
+    STATE_UNKNOWN,
+    STATE_LOCKED, STATE_UNLOCKED)
+
+_LOGGER = logging.getLogger(__name__)
+ATTR_CODE = 'code'
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """ Sets up the Verisure platform. """
+
+    if not verisure.MY_PAGES:
+        _LOGGER.error('A connection has not been made to Verisure mypages.')
+        return False
+
+    locks = []
+
+    locks.extend([VerisureDoorlock(value)
+                  for value in verisure.LOCK_STATUS.values()
+                  if verisure.SHOW_LOCKS])
+
+    add_devices(locks)
+
+
+# pylint: disable=abstract-method
+class VerisureDoorlock(LockDevice):
+    """ Represents a Verisure doorlock status. """
+
+    def __init__(self, lock_status, code=None):
+        self._id = lock_status.id
+        self._state = STATE_UNKNOWN
+        self._code = code
+
+    @property
+    def name(self):
+        """ Returns the name of the device. """
+        return 'Lock {}'.format(self._id)
+
+    @property
+    def state(self):
+        """ Returns the state of the device. """
+        return self._state
+
+    @property
+    def code_format(self):
+        """ Six digit code required. """
+        return '^\\d{%s}$' % verisure.CODE_DIGITS
+
+    def update(self):
+        """ Update lock status """
+        verisure.update_lock()
+
+        if verisure.LOCK_STATUS[self._id].status == 'unlocked':
+            self._state = STATE_UNLOCKED
+        elif verisure.LOCK_STATUS[self._id].status == 'locked':
+            self._state = STATE_LOCKED
+        elif verisure.LOCK_STATUS[self._id].status != 'pending':
+            _LOGGER.error(
+                'Unknown lock state %s',
+                verisure.LOCK_STATUS[self._id].status)
+
+    @property
+    def is_locked(self):
+        """ True if device is locked. """
+        return verisure.LOCK_STATUS[self._id].status
+
+    def unlock(self, **kwargs):
+        """ Send unlock command. """
+        if ATTR_CODE in kwargs:
+            self._code = kwargs[ATTR_CODE]
```

Comment:
By caching the code you are allowing people to lock and unlock this lock without providing a code after their providing the code once for `unlock` or `lock.

This should not be the default behavior as it reduces security for people and they might not be aware of it. I am fine by allowing this behavior using a config flag.

When a code has been cached, the `code_format` property should return `None`.

Should the code cache be invalidated if the code was wrong?